### ISSUE TITLE
No webhook for voice messages

### ISSUE THREAD

**Issue by andreydruz** [asternic/wuzapi#31]
Hello,

I'm trying to read voice messages, but the webhook doesn't work on them.

Is it by design, or did I make some installation or configuration errors?

```
2024-03-27T18:18:43Z INF Message Received | role=server host=0.0.0.0 id=E681F395EBE43BD9463D1BD84C8154E8 source=******@s.whatsapp.net parts=pushname: ******, timestamp: 2024-03-27 18:18:43 +0000 UTC, type: media

2024-03-27T18:19:50Z INF Chat Presence received | role=server host=0.0.0.0 state=composing media=audio chat=******@s.whatsapp.net sender=******@s.whatsapp.net

2024-03-27T18:19:50Z WRN Skipping webhook. Not subscribed for this type | role=server host=0.0.0.0 type=ChatPresence

2024-03-27T18:19:52Z INF Chat Presence received | role=server host=0.0.0.0 state=paused media= chat=******@s.whatsapp.net sender=******@s.whatsapp.net

2024-03-27T18:19:52Z WRN Skipping webhook. Not subscribed for this type | role=server host=0.0.0.0 type=ChatPresence

2024-03-27T18:19:53Z INF Message Received | role=server host=0.0.0.0 id=12D1A7116861B1EB87FB7EF1943E8B86 source=******@s.whatsapp.net parts=pushname: ******, timestamp: 2024-03-27 18:19:53 +0000 UTC, type: media

2024-03-27T18:19:59Z INF Chat Presence received | role=server host=0.0.0.0 state=composing media=audio chat=******@s.whatsapp.net sender=******@s.whatsapp.net

2024-03-27T18:19:59Z WRN Skipping webhook. Not subscribed for this type | role=server host=0.0.0.0 type=ChatPresence

2024-03-27T18:20:02Z INF Chat Presence received | role=server host=0.0.0.0 state=paused media= chat=******@s.whatsapp.net sender=******@s.whatsapp.net

2024-03-27T18:20:02Z WRN Skipping webhook. Not subscribed for this type | role=server host=0.0.0.0 type=ChatPresence

2024-03-27T18:20:02Z INF Message Received | role=server host=0.0.0.0 id=1ABDEE55801B3CBB35209528F5085616 source=******@s.whatsapp.net parts=pushname: ******, timestamp: 2024-03-27 18:20:02 +0000 UTC, type: media
```

**Comment by asternic** [comment]
If you mean PTT audios, they will fire a webhook as any other message type. The error you posted is because of not being subscribed to ChatPresence, but those particular messages are NOT the actual audio message being sent, but the intermediate statuses while the audio is being recorded. You can subscribe to All or some events on /session/connect. Seems you did not subscribe to ChatPresence, you can add that if you want to get those chatpresence statues (user is recording audio, user is typing, etc)

**Comment by ruben18salazar3** [comment]
Tampoco consigo recibir las notas de voz mediante los webhooks, solo recibo los eventos de composing/paused.

Esto solo acontece para las notas de voz, cuando intento recibir imágenes, videos, documentos, llegan sin problemas.

Será un problema general??

**Comment by lordacer0** [comment]
Good morning @asternic , after several tests the PTT audios do not launch a webhook unfortunately. Is it possible to review it? Thank you very much, excellent project.

**Comment by lordacer0** [comment]
> Tampoco consigo recibir las notas de voz mediante los webhooks, solo recibo los eventos de composing/paused.
> 
> Esto solo acontece para las notas de voz, cuando intento recibir imágenes, videos, documentos, llegan sin problemas.
> 
> Será un problema general??

Good evening, did you find a solution to the webhook problem with the audios?

**Comment by asternic** [comment]
Just tried, works fine for me. I cannot say how general the problem might be. In any case, five days ago I updated wuzapi to use the latest wahtsmeow version, so you might want to update if you didn't already.

**Comment by lordacer0** [comment]
> Just tried, works fine for me. I cannot say how general the problem might be. In any case, five days ago I updated wuzapi to use the latest wahtsmeow version, so you might want to update if you didn't already.
![image](https://github.com/user-attachments/assets/bcd75026-5a59-4066-974d-2ab3f70252bd)

I'm going to recreate the image, I'm using docker, but I don't think that's the problem. But I'm going to keep looking now that you confirm that you currently don't have any problems and it works for you. Thanks for responding.

**Comment by JuniorPaula** [comment]
I was facing the same problem, where the webhook was not called with the audio data. After debugging the code I noticed that the program failed when trying to save the received audio file. The failure occurred specifically on the line where the file path was generated based on the extension obtained from the MIME type. The log revealed that the mime.ExtensionsByType function, used to determine the file extension, was returning an empty array (exts: []), despite there being no explicit error.

This behavior resulted in an "index out of range" error, as the code attempted to access exts[0], which did not exist.

The solution I found was to add a check to ensure the extensions array is not empty before accessing the first element. In case the array is empty, I defined a default extension for Ogg audio files (.ogg). This ensures that the file path is correctly assembled regardless of the result of `mime.ExtensionsByType`.
```go
var ext string
if len(exts) > 0 {
    ext = exts[0]
} else {
    ext = ".ogg"
}
path = filepath.Join(userDirectory, evt.Info.ID+ext)
```

This worked for me.
